### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Deeds67** [vercel/next.js#56614]
### Link to the code that reproduces this issue

https://github.com/Deeds67/nextjs-max-call-stack-size-exceeded

### To Reproduce

1. Add dependency `"@material-tailwind/react": "^2.1.0",`
2. Create a `NavbarDefault` component as listed here: https://www.material-tailwind.com/docs/react/navbar#navbar
3. Render this component on a page
4. Run `npm run build`

### Current vs. Expected behavior

Current behavior:  

Build fails with the following error:
```RangeError: Maximum call stack size exceeded```

Expected behavior: Build should successfully pass.

Here's the full error output:

```
✗ npm run build

> max-call-stack-error@0.1.0 build
> next build

 ✓ Creating an optimized production build    
 ✓ Compiled successfully
 ✓ Linting and checking validity of types    
 ✓ Collecting page data    
   Generating static pages (0/5)  [    ]RangeError: Maximum call stack size exceeded
    at P (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:231722)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:227295)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228496
    at Array.forEach (<anonymous>)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228469
    at Array.forEach (<anonymous>)
    at a (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228289)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228553
    at Array.forEach (<anonymous>)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228246)

Error occurred prerendering page "/_not-found". Read more: https://nextjs.org/docs/messages/prerender-error
RangeError: Maximum call stack size exceeded
    at P (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:231722)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:227295)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228496
    at Array.forEach (<anonymous>)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228469
    at Array.forEach (<anonymous>)
    at a (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228289)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228553
    at Array.forEach (<anonymous>)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228246)
RangeError: Maximum call stack size exceeded
    at P (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:231722)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:227295)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228496
    at Array.forEach (<anonymous>)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228469
    at Array.forEach (<anonymous>)
    at a (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228289)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228553
    at Array.forEach (<anonymous>)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228246)

Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
RangeError: Maximum call stack size exceeded
    at P (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:231722)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:227295)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228496
    at Array.forEach (<anonymous>)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228469
    at Array.forEach (<anonymous>)
    at a (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228289)
    at /home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228553
    at Array.forEach (<anonymous>)
    at l (/home/pierre/dev/ts/max-call-stack-error/.next/server/chunks/241.js:5:228246)
 ✓ Generating static pages (5/5) 

> Export encountered errors on following paths:
	/_not-found
	/page: /
```

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #202309191142~1695998943~22.04~070916d SMP PREEMPT_DYNAMIC Fri S
Binaries:
  Node: 20.3.1
  npm: 9.6.7
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 13.5.5-canary.4
  eslint-config-next: 13.5.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Additional context

App works when running `npm run dev`, but not when building a production ready app using `npm run build`

<sub>[WEB-1744](https://linear.app/vercel/issue/WEB-1744/maximum-call-stack-size-exceeded)</sub>

**Comment by balazsorban44** [comment]
Thanks, I believe this was a regression that slipped in in https://github.com/vercel/next.js/releases/tag/v13.5.4-canary.9 via https://github.com/vercel/next.js/pull/56281

We will have a look!

**Comment by sheldon-welinga** [comment]
I think its an issue related to `@material-tailwind/react` library. If you remove it. Your project build's properly

**Comment by Deeds67** [comment]
I managed to work around the issue by reverting to an older version: `"next": "13.5.3"`. Bug appears in `13.5.4` and later.

**Comment by kdy1** [comment]
This is a regression between `swc_core@v0.83.12` and `swc_core@0.83.28`

https://github.com/vercel/next.js/blob/76e06d55a8c42090dafda25ec31d9eeaa7ec647c/Cargo.lock#L6661-L6663

https://github.com/vercel/next.js/blob/1e8dca4a87113f9d6ce0505c90f9a3d89096cefd/Cargo.lock#L5811-L5813

**Comment by kdy1** [comment]
I think this can be a same issue as https://github.com/vercel/next.js/issues/56550
I'll investigate

**Comment by kdy1** [comment]
<img width="1840" alt="image" src="https://github.com/vercel/next.js/assets/29931815/90ad532b-4772-4d80-b8ff-11e77b75dae8">

This is the same issue as `exceljs` issue linked at https://github.com/vercel/next.js/issues/56550#issuecomment-1751809466

**Comment by alexews** [comment]
Hello,
I've faced same issue with Nextjs 13.5.4

**Comment by kelihansen** [comment]
I'm still experiencing this issue with v13.5.6-canary.2 of next 🤔

**Comment by kdy1** [comment]
@kelihansen If so, file another issue. It's a different one.

**Comment by NPJigaK** [comment]
When using Material Tailwind, downgrading the next version to 13.5.3 solves the problem. 13.5.4 or higher will cause an error. v13.5.7-canary.9 as well.

**Comment by rjnls** [comment]
> When using Material Tailwind, downgrading the next version to 13.5.3 solves the problem. 13.5.4 or higher will cause an error. v13.5.7-canary.9 as well.

Thank you @NPJigaK, my issue was not during build, instead it was the same error at runtime in the browser console. Downgrade to 13.5.3 solved it. I couldn't even use a component that consisted of nothing but <Card>Hi</Card> without this happening. All of this worked perfectly in local run dev and I could always build fine, never once in 4 hours of debugging did it fail to build (except for typos), only deployment revealed the issue and only during runtime. That had me dissecting the chunk arrays in the build output, aka hell.

**Comment by kdy1** [comment]
@NPJigaK Does it happen with v13.5.7-canary.16?

**Comment by kelihansen** [comment]
v13.5.7-canary.16 appears to fix the issue for me 🙌

**Comment by NPJigaK** [comment]
@kdy1 It was fixed in v13.5.7-canary.16! Thanks!
